src/mesh/replicated_mesh.C

```diff
@@ -1417,5 +1417,151 @@ dof_id_type ReplicatedMesh::n_active_elem () const
                                                  this->active_elements_end()));
 }
 
+std::map<Point, std::vector<std::vector<Point>>>
+ReplicatedMesh::get_boundary_points() const
+{
+  // find number of disconnected subdomains
+  std::vector<Point> interior_points;
+  std::vector<subdomain_id_type> subdomains(max_elem_id() + 1, Elem::invalid_subdomain_id);
```

Comment:
Why copy subdomain_id() values into a vector?  They don't change during this function and you always have easy access to the element to query them directly.

Reply:
Oh, never mind, you're tracking unvisited elements.

Reply:
Still kind of confusing.  I'd have either used a vector<bool> or an unordered_set.  Neither is so much better as to make it worth changing this though.

Reply:
No, I'm misunderstanding again.  You're not using subdomain_id(), you're defining your own subdomains based on mesh connectivity?

I should probably stop the stream-of-consciousness here until I'm finished with the readthrough.

Reply:
I should have added an inline comment on this variable like `markers for all elements to indicate if they have been visited`.

Reply:
A comment that these subdomain ids are unrelated to the subdomain_id() would have been helpful.  This seems obvious on closer reading but some times people like me need help seeing the obvious.  :-)

Reply:
Will add that into the inline comment.